algebra/group/subgroup.lean

```diff
@@ -0,0 +1,147 @@
+import data.set.basic init.function data.equiv init.logic 
+import algebra.group
+
+open set
+
+universes u v
+variables {G : Type u} {H : Type v}
+
+class subgroup [group G] (S : set G) : Prop := 
```

Comment:
https://github.com/leanprover/mathlib/blob/master/group_theory/subgroup.lean

Reply:
Thanks Kenny for pointing this out: @johoelzl did this after Mitchell had finished for the summer, and neither of us had seen it in the meantime.

Rather than immediately update this PR so that it uses this code, I'll send an intermediate commit that modifies the existing `subgroup.lean` so that it uses `is_subgroup` as a typeclass. After everyone's had a chance to comment on that, I'll "rebase" Mitchell's work onto Johannes's.